src/helpers.rs

```diff
@@ -1,14 +1,26 @@
+#[cfg(any(feature = "html_root_url", feature = "markdown"))]
 use std::fmt::Display;
+#[cfg(any(feature = "html_root_url", feature = "markdown",
```

Comment:
These carefully crafted conditional imports seem like they'll be a nightmare to maintain.

I think we could get rid of a bunch of them by moving as much as possible into smaller scopes. Ideally, I would like to keep the conditional parts as small and self-contained as possible. The less moving parts, the better.

Reply:
Most of these have been moved.